pkg/runtime/dispatcher.go

```diff
@@ -103,43 +114,61 @@ type dispatcher struct {
 	// It is not directly used by dispatcher.
 	mapper expr.Evaluator
 
-	// <resolver>
-	resolver atomic.Value
-
 	// gp is used to dispatch multiple adapters concurrently.
 	gp *pool.GoroutinePool
+
+	// protects resolver
+	sync.RWMutex
+
+	// <resolver>
```

Comment:
Just remove this comment, it doesn't add anything and lets you collapse the mutex beside the resolver.

```go
m sync.WRMutex
r Resolver
```